TN083: Initial commit on fa lag error analysis

b1quint opened the pull request.

b1quint committed:
TN083: Initial commit on fa lag error analysis

craiglagegit committed:
Initial push of delay analysis notebook

b1quint committed:
TN083: Update notebook with single bumpt test analysis

b1quint committed:
Run black, update docstrings and type annotations

b1quint committed:
TN083: Add statistical analysis for lags

b1quint committed:
TN083: remove duplicated notebook

craiglagegit committed:
Adding missing pdf.close() statement.

craiglagegit commented:
I reviewed the notebook and the Python file.  They look good with one exception.  The PDF file when running all of the actuators needs a pdf.close() statement at the end.  Otherwise the resulting PDF file is unreadable.  I corrected this and pushed the new version.  From my standpoint it is ready for review.

craiglagegit committed:
Pushing changes to allow delay analysis for both bump tests and slews.

craiglagegit commented:
Hi @b1quint .  I modified the notebook and the Python file you created so that it supports actuator delays from both bump tests and slews, as was requested.  So I changed the name of the Python file from bump_test_analysis.py to actuator_analysis.py.  I'd like your comments on what I have done.  We might want to re-structure things, since it is now better associated with SITCOMTN-107 than SITCOMTN-083.

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOMTN-083_fa_error_lag_analysis.ipynb:
This file has a couple of broken cells. The first cell inside the "Analyze all the Force actuators" session has a call to `bta.plot_bump_test_actuator_delay` which seems to be replaced with `act.plot_actuator_delay`. This is easy to miss because it lives inside a generic `try/except`, so we will never get those plots.

b1quint commented on notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOMTN-083_fa_error_lag_analysis.ipynb:
The cell inside "Delay Histograms" is also broken. It still calls `bta` instead of `act`. Plus, `id` is the force actuator id, not the index array. This breaks the cell in line 8.

b1quint commented:
I see that `m1m3.py` has conflicts with the version in the `develop` branch. It's probably because of the black format. I will have a look.

b1quint committed:
Address comments in PR#104

- Solve conflict with origin/develop
- Fix calls to renamed module
- Move text in histograms to the plot title

craiglagegit commented:
Hi Bruno.  Sorry about that, I didn't actually test those cells, so of course there was an error.  I ran the whole notebook this time and it looks good.  Ready to merge from my standpoint.

craiglagegit reviewed (approved):
Ready to merge.
